resources/install.rb

```diff
@@ -0,0 +1,160 @@
+property :rs_account_pwd, kind_of: String
+property :rs_startup, kind_of: String, equal_to: ['Automatic', 'Manual', 'Disabled', 'Automatic (Delayed Start)'], default: 'Automatic'
+property :rs_mode, kind_of: String, default: 'FilesOnlyMode'
+property :as_sysadmins, kind_of: [Array, String], default: ['Administrator']
+property :sql_account, kind_of: String, default: 'NT AUTHORITY\NETWORK SERVICE'
+property :sql_account_pwd, kind_of: String
+property :browser_startup, kind_of: String, equal_to: ['Automatic', 'Manual', 'Disabled', 'Automatic (Delayed Start)'], default: 'Disabled'
+property :version, kind_of: String, default: '2012'
+property :source_url, kind_of: String
+property :package_name, kind_of: String
+property :package_checksum, kind_of: String
+property :installer_timeout, kind_of: Integer, default: 1500
+property :accept_eula, kind_of: [TrueClass, FalseClass], default: false
+property :product_key, kind_of: String
+property :update_enabled, kind_of: [TrueClass, FalseClass], default: true
+property :instance_name, kind_of: String, default: 'SQLEXPRESS'
+property :feature, kind_of: [Array, String], default: %w(SQLENGINE REPLICATION SNAC_SDK)
+property :install_dir, kind_of: String, default: 'C:\Program Files\Microsoft SQL Server'
+property :instance_dir, kind_of: String, default: 'C:\Program Files\Microsoft SQL Server'
+property :sql_data_dir, kind_of: String
+property :sql_backup_dir, kind_of: String
+property :sql_user_db_dir, kind_of: String
+property :sql_user_db_log_dir, kind_of: String
+property :sql_temp_db_dir, kind_of: String
+property :sql_temp_db_log_dir, kind_of: String
+property :filestream_level, kind_of: Integer, equal_to: [0, 1, 2, 3], default: 0
+property :filestream_share_name, kind_of: String, default: 'MSSQLSERVER'
+property :sql_collation, kind_of: String
+property :dreplay_ctlr_admins, kind_of: [Array, String], default: ['Administrator']
+property :dreplay_client_name, kind_of: String
+
+action :install do
+  if new_resource.feature.include?('DREPLAY_CLT') && new_resource.dreplay_client_name.nil?
+    ::Chef::Application.fatal!('You cannot select include the DREPLAY_CLT feature with specifing a controller to target.')
+  end
+
+  if new_resource.security_mode == 'Mixed Mode Authentication' && new_resource.sa_password.nil?
+    ::Chef::Application.fatal!('You cannot have set the security mode to "Mixed Mode Authenication" without specifying the sa_password property')
+  end
+
+  config_file_path = ::File.join(Chef::Config[:file_cache_path], 'ConfigurationFile.ini')
+
+  x86_64 = node['kernel']['machine'] =~ /x86_64/
+
+  sql_sys_admin_list = if new_resource.sysadmins.is_a? Array
+                         new_resource.sysadmins.map { |account| %("#{account}") }.join(' ') # surround each in quotes, space delimit list
+                       else
+                         %("#{new_resource.sysadmins}") # surround in quotes
+                       end
+
+  as_sys_admin_list = if new_resource.as_sysadmins.is_a? Array
```

Comment:
These are good candidates for small helper methods in the action_class of this resource. It'll ease readability of the code